examples/platforms/emsk/random.c

```diff
@@ -0,0 +1,101 @@
+/*
+ *  Copyright (c) 2016, The OpenThread Authors.
+ *  All rights reserved.
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions are met:
+ *  1. Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *  2. Redistributions in binary form must reproduce the above copyright
+ *     notice, this list of conditions and the following disclaimer in the
+ *     documentation and/or other materials provided with the distribution.
+ *  3. Neither the name of the copyright holder nor the
+ *     names of its contributors may be used to endorse or promote products
+ *     derived from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ *  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ *  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ *  ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ *  LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ *  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ *  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ *  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ *  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ *  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/**
+ * @file
+ *   This file implements a pseudo-random number generator.
+ *
+ * @warning
+ *   This implementation is not a true random number generator and does @em satisfy the Thread requirements.
+ *
+ */
+
+#include <common/code_utils.hpp>
+
+#include "openthread/platform/random.h"
+#include "openthread/platform/radio.h"
+#include "platform-emsk.h"
+
+#include <stdlib.h>
+
+#include <stdio.h>
+
+static unsigned int seed;
+
+void emskRandomInit(void)
+{
+    unsigned int i;
+
+    printf("Node No.:");
+    scanf("%d", &i);
+    printf("%d\n", i);
+    seed = 10 + i;
+    srand(seed);
```

Comment:
Thanks for your clear response and suggestion.
1) As for ARC, we provide processor solution but not chips. So EMSK will not used in any end product , such as smart temperature sensor. And unlike chips, there is less data sheet docs for ARC and EMSK in public. For application development, I recommend www.embarc.org. For more details of EMSK, please download source code and see the doc generated by doxygen. We will open it to github soon. 
2) Yes. The best way is to generater a true random or quasi-random number. Unfortunately I can't get the FPGA ID by software. We can regenerate the core image to add a number in hardware. But it is not a generator and the image can not be released.
3) We are glad to see there will be a end product with using a chip powered by OpenThread + mbedTLS + ARC + ARC tools. I am sure the chip must contain a random number generator as required by Thread. But There is no native Thread hardware support on EMSK. It is a hardware platform for verification and demonstration.
I am trying to find a better way to implement it on EMSK. I will update it soon. Thanks again.